packages/@dcl/inspector/src/components/Tree/Tree.tsx

```diff
@@ -36,142 +28,130 @@ type Props<T> = {
   onToggle: (value: T, isOpen: boolean) => void
 }
 
-type Tree<T> = Props<T> & ContextMenuProps<T>
-
 const getDefaultLevel = () => 1
 const getLevelStyles = (level: number) => ({ paddingLeft: `${level * 10}px` })
 const getExpandStyles = (active: boolean) => ({ height: active ? 'auto' : '0', overflow: 'hidden', display: 'block' })
 const getEditModeStyles = (active: boolean) => ({ display: active ? 'none' : '' })
 
-function Tree<T>(props: Tree<T>) {
-  const {
-    value,
-    level = getDefaultLevel(),
-    getId,
-    getChildren,
-    getLabel,
-    isOpen,
-    isSelected,
-    onSetParent,
-    canRename,
-    canAddChild,
-    canRemove,
-    canToggle,
-    onRename,
-    onAddChild,
-    onRemove,
-    onToggle,
-    contextMenu,
-    onContextMenuChange
-  } = props
-  const id = getId(value)
-  const label = getLabel(value)
-  const open = isOpen(value)
-  const selected = isSelected(value)
-  const enableRename = canRename ? canRename(value) : true
-  const enableAddChild = canAddChild ? canAddChild(value) : true
-  const enableRemove = canRemove ? canRemove(value) : true
-  const enableToggle = canToggle ? canToggle(value) : true
-  const [editMode, setEditMode] = useState(false)
-  const [insertMode, setInsertMode] = useState(false)
-  const [contextMenuPosition, setContextMenuPosition] = useState<Position | undefined>(undefined)
-
-  const canDrop = useCallback(
-    (target: T, source: T): boolean => {
-      if (getId(target) === getId(source)) return false
-      return getChildren(target).every(($) => canDrop($, source))
-    },
-    [getId, getChildren]
-  )
+function Tree<T>(_props: Props<T>) {
+  const Component = withContextMenu<Props<T>>((props) => {
+    const {
+      extraContextMenu,
```

Comment:
what do you think about renamig this to `getExtraContextMenu` in order to keep it consistent with the other `(value: T) => ...` getters in the props?

Reply:
sure!